doc/design/couler_sqlflow.md

```diff
@@ -20,4 +20,51 @@ The migration includes the following parts:
 
 1. Instead of having multiple codegens, let us have only one, `codegen_couler.go`, which translates the [Intermediate Representation](/doc/design/intermediate_representation.md)(IR) of a SQL program into a Couler program. Then, SQLFlow can run the Couler compiler to convert further and execute the workflow.
 
-For example, `codegen_couler.go` converts a `SELECT ... TO TRAIN` statement into the call to `sqlflow.couler.{xgboost,tensorflow.elasticdl}.train(train_ir)`, which `train_ir` is the SQLFlow IR with JSON format.
+For example, `codegen_couler.go` converts a `SELECT ... TO TRAIN` statement into the call to `sqlflow.couler.{xgboost,tensorflow.elasticdl}.train(...)`.
+
+## Couler Step Function
+
+For example, an extended SQL `SELECT ... TO TRAIN xgboost.booster ...` would be compiled into a
+an Couler step function by `codegen_couler.go`:
+
+``` python
+couler.sqlflow.run('echo "SELECT ... TO TRAIN xgboost.booster" | sqlflow -parse | python -m sqlflow_submitter.xgboost.train', image="sqlflow/sqlflow_submitter")
+```
+
+In the above Couler function:
+
+- `sqlflow -parse` is a command-line tool which parses an extended SQL into an SQLFlow IR with JSON format, like:
+
+    ``` json
+    {
+        "dataSource": "mysql://user:pass@192.168.1.1:3306",
+        "select": "SELECT * FROM iris.train",
+        "validation_select": "SELECT * FROM iris.test",
+        "estimator": "XGBOOST.gbtree",
+        "attributes": {"train.num_boost_round": 30},
+        "features": {"sepal_length": {"type": "numeric", "shape": [1], "field_meta":{"name":"sepal_length", "dtype": "float32", "delimiter": "", "is_sparse": False}}...},
+        "label": {"class": {"type": "numeric", "shape": [1], "field_meta": ...}}
```

Comment:
I am not sure if the JSON needs to be like SQLFlow TrainIR. Why not just a list of SQL statements?

Reply:
I'm sure the JSON is `TrainID` which is the input of `sqlflow_submitter.xgboost.train`.
SQLFLow command-line compiles the SQL into IR, the submitter program submit the estimator. We don't need to export Feature derivation/verify to Python code.

Reply:
As the discussion, `sqlflow -parse` outputs a protobuf text file.